src/main/java/org/tugraz/sysds/runtime/controlprogram/federated/FederatedWorkerHandler.java

```diff
@@ -265,45 +176,139 @@ else if (format.equalsIgnoreCase(FORMAT_TYPE_VALUE_CSV)) {
+		long varID = (Long) request.getParam(1);
+		return executeAggregation(varID, operator);
+	}
+
+	private FederatedResponse executeAggregation(long varID, AggregateUnaryOperator operator) {
+		Data dataObject = _vars.get(varID);
+		if (dataObject.getDataType() != Types.DataType.MATRIX) {
+			return new FederatedResponse(FederatedResponse.Type.ERROR,
+					"FederatedWorkerHandler: Aggregation only supported for matrices, not for "
+							+ dataObject.getDataType().name());
+		}
+		MatrixObject matrixObject = (MatrixObject) dataObject;
+		MatrixBlock matrixBlock = matrixObject.acquireRead();
+		// create matrix for calculation with correction
+		MatrixCharacteristics mc = new MatrixCharacteristics();
+		// find out the characteristics after aggregation
+		operator.indexFn.computeDimension(matrixObject.getDataCharacteristics(), mc);
+		// make outBlock right size
+		int outNumRows = (int) mc.getRows();
+		int outNumCols = (int) mc.getCols();
+		if (operator.aggOp.existsCorrection()) {
+			// add rows for correction
+			int numMissing = operator.aggOp.correction.getNumRemovedRowsColumns();
+			if (operator.aggOp.correction.isRows())
+				outNumRows += numMissing;
+			else
+				outNumCols += numMissing;
+		}
+		MatrixBlock ret = new MatrixBlock(outNumRows, outNumCols, operator.aggOp.initialValue);
+		try {
+			LibMatrixAgg.aggregateUnaryMatrix(matrixBlock, ret, operator);
+		}
+		catch (Exception e) {
+			return new FederatedResponse(FederatedResponse.Type.ERROR, "FederatedWorkerHandler: " + e);
+		}
+		// result block without correction
+		ret.dropLastRowsOrColumns(operator.aggOp.correction);
+		return new FederatedResponse(FederatedResponse.Type.SUCCESS, ret);
+	}
+
 	@SuppressWarnings("unused")
 	private FederatedResponse createMatrixObject(MatrixBlock result) {
 		MatrixObject resTo = new MatrixObject(Types.ValueType.FP64, OptimizerUtils.getUniqueTempFileName());
-		MetaDataFormat metadata = new MetaDataFormat(new MatrixCharacteristics(result.getNumRows(),
-			result.getNumColumns()), OutputInfo.BinaryBlockOutputInfo, InputInfo.BinaryBlockInputInfo);
+		MetaDataFormat metadata = new MetaDataFormat(
+				new MatrixCharacteristics(result.getNumRows(), result.getNumColumns()),
+				OutputInfo.BinaryBlockOutputInfo, InputInfo.BinaryBlockInputInfo);
 		resTo.setMetaData(metadata);
 		resTo.acquireModify(result);
 		resTo.release();
 		long result_var = _seq.getNextID();
 		_vars.put(result_var, resTo);
 		return new FederatedResponse(FederatedResponse.Type.SUCCESS, result_var);
 	}
-	
+
 	private static void checkNumParams(int actual, int... expected) {
-		if( Arrays.stream(expected).anyMatch(x -> x==actual) )
+		if (Arrays.stream(expected).anyMatch(x -> x == actual))
 			return;
-		throw new DMLRuntimeException(
-			"FederatedWorkerHandler: Received wrong amount of params:" + " expected="
+		throw new DMLRuntimeException("FederatedWorkerHandler: Received wrong amount of params:" + " expected="
```

Comment:
Ok I give up. It's a DMLRuntimeException 😜

Reply:
Yes